### PR TITLE
python-setup: rely on new `virtualenv` for venv creation in Ubuntu 22.04

### PR DESCRIPTION
**Pull request by RasmusWL**
Removes the hotfix from https://github.com/github/codeql-action/pull/1257

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

Have not updated either, since this should not be a user visible change

### CODE REVIEW

**Comment by RasmusWL** [comment]
Forgot to link to the release notes of `virtualenv`: https://virtualenv.pypa.io/en/latest/changelog.html#bugfixes-20-16-6 -- this was actually fixed some time ago, but apparently my automation for noticing a new release only kicked in for a major release :shrug:

File: python-setup/install_tools.sh [hunk 1]
```diff
@@ -45,7 +44,7 @@ if command -v python2 >/dev/null 2>&1; then
 		curl --location --fail https://bootstrap.pypa.io/pip/2.7/get-pip.py | python2
 	fi
 
-	python2 -m pip install --user --upgrade pip 'setuptools<60' wheel
+	python2 -m pip install --user --upgrade pip setuptools wheel
 
 	python2 -m pip install --user 'virtualenv!=20.12.0'
```
Comment (henrymercer) [inline python-setup/install_tools.sh:49]: Does it matter which version we install here?

**Review by henrymercer — COMMENTED** [review]
Thanks for following up with the cleanup!

File: python-setup/install_tools.sh (hunk 1 above)
Comment (RasmusWL) [reply python-setup/install_tools.sh:49]: The change in this PR does not affect Python 2.

context: Latest version of `virtualenv` that I'm allowed in install for Python 2 is `20.15.1`, I did check this locally, but failed to put it in the description, so good question :+1:

File: python-setup/install_tools.sh [hunk 2]
```diff
@@ -12,16 +12,15 @@ export PATH="$HOME/.local/bin:$PATH"
...
-# virtualenv is a bit nicer for setting up virtual environment, since it will provide up-to-date versions of
-# pip/setuptools/wheel which basic `python3 -m venv venv` won't
-python3 -m pip install --user virtualenv
+# virtualenv is a bit nicer for setting up virtual environment, since it will provide
+# up-to-date versions of pip/setuptools/wheel which basic `python3 -m venv venv` won't.
+#
+# version 20.16.5 (Python 3 only) had some problems when used together with newer
+# versions of setuptools (60+) and would not always put binaries under `<venv-path>/bin`
+# -- see https://github.com/github/codeql-action/issues/1249 for more details.
+python3 -m pip install --user --upgrade 'virtualenv>20.16.5'
```
[the remark this answers is not in the document]
Comment (RasmusWL) [reply python-setup/install_tools.sh:23]: I haven't seen any evidence that this should have affected Windows. So I think it's fine as-is, but good question :+1:

**Review by henrymercer — APPROVED** [review]

**Pull request merged**